Updated DRF and Python version combinations

kevin-brown opened the pull request:
This should be supported out of the box, but let's find out...

* [x] Get Django 2.1 passing
* [x] Backfill Python 3.5 support across DRF versions

By the end of this we are going to be at over a hundred different supported combinations. We may want to look into dropping really old combinations from the testing matrix, or cutting some of the middle versions since we don't have much (if any) special casing going on.

jpadilla commented:
@kevin-brown I'd definitely drop anything pre Django 1.11 at this point and perhaps just test against latest stable DRF release. That would simplify the matrix considerably.

kevin-brown commented:
Once again, no code changes were required to support the newer versions (latest DRF, latest Django, Python up to 3.6). I dropped support for Django 1.6/1.7 and Python 3.3 because it made it easier to set up the Tox matrix. I also switched to tox-travis to make specifying that matrix in Travis much less painful.

jpadilla commented:
@kevin-brown thank you! 🎉
